meta-refkit/classes/image-installer.bbclass

```diff
@@ -0,0 +1,370 @@
+# This class can be inherited by an arbitrary image recipe to install
+# the "image-installer" command plus one or more image files into the
+# image.
+#
+# Basically that turns the image using image-installer.bbclass into an
+# "installer image" which can be booted from a removal media to
+# install those other images permanently onto internal storage of a
+# device.
+#
+# The "image-installer" is a shell script that gets assembled from
+# several fragments provided by image-installer.bbclass, by the distro
+# or by the image recipe. The expected usage is that distros using
+# image-installer.bbclass will heavily customize the final script.
+#
+# Because the shell code is embedded in the recipe, regular bitbake
+# variables can be injected easily and the shell syntax is checked
+# already by the bitbake parser. The downside is that we are limited
+# to shell code that the bitbake parser understands (no bash
+# extensions, for example) and error reports about the shell syntax
+# are a bit hard to read.
```

Comment:
How would one write tests if it was a separate file? I don't think that would be different. In both cases one needs to boot into the installer image and then exercise the script in various ways. This can be done via oe-selftest.

Whether embedding vs. separate file is easier from a maintenance perspective is debatable. I found it quite useful.